bluez/audio/main.c

```diff
@@ -45,9 +45,40 @@
 #include "headset.h"
 #include "manager.h"
 #include "gateway.h"
+#include "wakelock.h"
+#include "main.h"
+
+#define AUDIO_WAKELOCK_DURATION 5
 
 static GIOChannel *sco_server = NULL;
 
+static struct wakelock *audio_wakelock = NULL;
+
+static guint audio_wakelock_source = 0;
+
+static gboolean audio_wakelock_put(gpointer user_data)
+{
+	wakelock_release(audio_wakelock);
+	audio_wakelock_source = 0;
+	return FALSE;
+}
+
+void audio_wakelock_get(void)
+{
+	guint old_source = audio_wakelock_source;
+
+	audio_wakelock_source = g_timeout_add_seconds(AUDIO_WAKELOCK_DURATION,
+							audio_wakelock_put,
+							NULL);
+	if (audio_wakelock_source)
+		wakelock_acquire(audio_wakelock);
+
+	if (old_source) {
+		g_source_remove(old_source);
+		wakelock_release(audio_wakelock);
+	}
+}
```

Comment:
This looks a bit odd. I'm assuming the intent is to take wakelock on 1st call & install timer; renew the timer on subsequent calls. Now AFAICT audio_wakelock_get() x 2 -> timer active, but wakelock released.

Perhaps something like this?

if( !audio_wakelock_source )
  wakelock_acquire(audio_wakelock);
else
  g_source_remove(audio_wakelock_source);
audio_wakelock_source = g_timeout_add_seconds(...)

(plus release on glib timer failure, if that was the intention)